x/trigger/keeper/trigger_dispatcher.go

```diff
@@ -0,0 +1,97 @@
+package keeper
+
+import (
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+
+	types "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/x/group/errors"
+)
+
+const (
+	MaximumActions  uint64 = 5
+	MaximumQueueGas uint64 = 2000000
+)
+
+// ProcessTriggers Reads triggers from queues and attempts to run them.
+func (k Keeper) ProcessTriggers(ctx sdk.Context) {
+	var actionsProcessed uint64
+	var gasConsumed uint64
+
+	for !k.QueueIsEmpty(ctx) && actionsProcessed < MaximumActions {
+		item := k.QueuePeek(ctx)
+		triggerID := item.GetTrigger().Id
+		gasLimit := k.GetGasLimit(ctx, triggerID)
+
+		if gasLimit+gasConsumed > MaximumQueueGas {
+			return
+		}
+		actionsProcessed++
+		gasConsumed += gasLimit
+
+		k.Dequeue(ctx)
+		k.RemoveGasLimit(ctx, triggerID)
+
+		actions := item.GetTrigger().Actions
+		k.runActions(ctx, gasLimit, actions)
+	}
+}
+
+// RunActions Runs all the actions and constrains them by gasLimit.
+func (k Keeper) runActions(ctx sdk.Context, gasLimit uint64, actions []*types.Any) {
+	cacheCtx, flush := ctx.CacheContext()
+	gasMeter := sdk.NewInfiniteGasMeter()
```

Comment:
When do the gas costs actually get deducted from the user's account? Are we just always gonna use `1905nhash` for the conversion?

I guess maybe it'd work to just add all the gas limits to the gas meter during execution of the `CreateTrigger` so they're paid for up front using that Tx's gas prices.

Reply:
Nevermind. Just found where it's getting charged.